src/classes/i18n.js

```diff
@@ -83,42 +82,50 @@ export default class i18n {
         this._get_languages();
         this._load_cache();
 
-        return this;
-    }
+        this.loaded = true;
 
-    /**
-     * sets the language to use in translate function
-     * @public
-     * @param {string} lang
-     * @example
-     * i18n.set('zh_TW')
-     * @return {i18n} itself
-     */
-    set (lang = 'en') {
-        this.lang = lang;
         return this;
     }
 
     /**
      * gets the translation for the key and replaces the variables
      * @public
+     * @params {string}    language
      * @params {string}    key
      * @params {object}    variables
      * @return {string}    translation
      */
-    trans (key, variables = {}) {
+    trans (lang, key, variables = {}) {
+
+        if (!this.loaded) {
+            throw new Error('Language is not yet loaded. Please call .load() first');
+        }
+
 
         let default_lang = this.config.get('default');
-        let str = '';
 
-        if (this.translations[this.lang]
-         && this.translations[this.lang][key]) {
-            str = this.translations[this.lang][key];
-        }
-        else {
-            str = this.translations[default_lang][key] || '';
+
+        if (typeof variables === 'undefined') {
```

Comment:
Well, since you don't want to force the breaking change, please make sure you list out all the libraries and create tickets for them to be fixed...

There's really no use case where we would want to use the fallback without lang.